File: python/program/Ram.py
import Komponen

class Ram(Komponen):
    def __init__(self, kapasitor_gb=0, ddr='', merk='', nama=''):
        super().__init__(merk, nama)
        self.kapasitor_gb = kapasitor_gb
        self.ddr = ddr
    
    def set_kapasitor_gb(self, kapasitor_gb):
        self.kapasitor_gb = kapasitor_gb
    
    def set_ddr(self, ddr):
        self.ddr = ddr
    
    def get_kapasitor_gb(self):
        return self.kapasitor_gb
    
    def get_ddr(self):
        return self.ddr

File: cpp/program/Ram.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Ram : public Komponen{
private:
    int kapasitasGB;
    string ddr;
public:
    Ram(){}

    Ram(int kapasitasGB, string ddr, string merk, string nama) : Komponen(merk, nama){
        this->kapasitasGB = kapasitasGB;
        this->ddr = ddr;
    }

    void setkapasitasGB(int kapasitasGB){
        this->kapasitasGB = kapasitasGB;
    }
    void setddr(string ddr){
        this->ddr = ddr;
    }

    int getkapasitasGB(){
        return this->kapasitasGB;
    }
    string getddr(){
        return this->ddr;
    }

    ~Ram(){}
};